lib/srv/desktop/rdp/rdpclient/src/rdpdr/tdp.rs

```diff
@@ -341,6 +341,38 @@ pub struct SharedDirectoryDeleteRequest {
     pub path: UnixPath,
 }
 
+impl SharedDirectoryDeleteRequest {
+    /// Converts this request into a [`SharedDirectoryDeleteRequest`].
+    ///
+    /// Returns a tuple containing the [`SharedDirectoryDeleteRequest`] and a [`CString`],
+    /// which is the memory backing the [`SharedDirectoryDeleteRequest::path`] field.
+    /// It is the caller's responsibility to ensure that the [`CString`] lives until
+    /// the [`SharedDirectoryDeleteRequest::path`] is copied into Go-owned memory.
+    ///
+    /// See the example for [`SharedDirectoryCreateRequest`]'s `into_cgo`.
+    pub fn into_cgo(self) -> PduResult<(CGOSharedDirectoryDeleteRequest, CString)> {
```

Comment:
This is unsafe and very easy to misuse, see https://github.com/gravitational/teleport/pull/34011#discussion_r1380493631